### PR TITLE
Update not equals rule to check for tag existence

### PR DESCRIPTION
**Pull request by rmarianski**
Connects to https://github.com/tilezen/vector-datasource/issues/1043 and https://github.com/tilezen/vector-datasource/pull/1036.

@zerebubuth could you review please?

### CODE REVIEW

**Comment by zerebubuth** [comment]
:+1: looks good to me

**Comment by zerebubuth** [comment]
:+1:

**Comment by rmarianski** [comment]
@zerebubuth I liked your `coalesce` idea, and updated the code accordingly in f9fd38b.

File: data/migrations/create-sql-functions.py [hunk 1]
```diff
@@ -113,15 +113,14 @@ def create_column(self, name):
...
-        return '(%s) AND (%s)' % (exists_check, check)
-    else:
-        return check
+def ensure_sql_boolean(column, check, null_value):
+    # tags->'foo' is NULL when the hstore doesn't contain that key,
+    # which makes all the following comparisons NULL, so use coalesce
+    # to ensure we never evaluate to NULL
+    if column._is_tag:
+        sql_on_null = 'TRUE' if null_value else 'FALSE'
+        check = 'COALESCE(%s, %s)' % (check, sql_on_null)
```
Comment (zerebubuth) [inline data/migrations/create-sql-functions.py]: Does this need an 'else' branch in case the column isn't a tag?

Comment (zerebubuth) [inline data/migrations/create-sql-functions.py]: Oh, wait, no I see how this works - it just returns the `check` unmodified.

**Comment by rmarianski** [comment]
Whoops, we actually want the sense on the not equals check reversed, ie evaluate to false if the tag doesn't exist. Updated in 82f202c.

**Comment by zerebubuth** [comment]
Hmm.. So we don't have the meaning that `(A ≠ B) ≡ ¬(A = B)`? Our "not equals" really means "(A is not null and not equal to B) or (A is null)"?

**Comment by rmarianski** [comment]
> Hmm.. So we don't have the meaning that (A ≠ B) ≡ ¬(A = B)? Our "not equals" really means "(A is not null and not equal to B) or (A is null)"?

Yea, currently. I did it that way to match the intent in the yaml. But it's used very rarely, so it might have been better to update the usages in the yaml file instead and revert back the not equals meaning. What do you think?

**Comment by rmarianski** [comment]
Or we can also just get rid of not equals and replace the usages with the more generic `not` operator.

**Comment by zerebubuth** [comment]
The only use I see now (that the underground use has gone) is:

```yaml
  - filter: {disused: -no}
```

Which does really mean "disused is present but not 'no'". Should we change that to:

```yaml
  - filter:
    - disused: True
    - not: { disused: no }
```

?

I think I agree with you that it's preferable, as it's clearer about the case where `disused` is NULL.

**Comment by rmarianski** [comment]
Agreed, updated in acc79b6.

**Comment by zerebubuth** [comment]
:+1: looks good to me!

**Pull request merged**